src/model/AssetsCompiler.ts

```diff
@@ -316,29 +316,33 @@ export class AssetsCompiler {
         }
     }
 
-    private static findPluginPath(repositoryDir: string, pluginName: string, repoDependencies: string[]): string {
-        let relativePath = pluginName;
-        if (fs.existsSync(path.join(repositoryDir, relativePath))) {
-            return path.join(repositoryDir, relativePath);
+    public static findPluginPath(repositoryDir: string, pluginName: string, repoDependencies: string[]): string {
+        let relativePathToPlugin = pluginName;
+        if (AssetsCompiler.isPluginFolder(repositoryDir, pluginName)) {
+            return path.join(repositoryDir, relativePathToPlugin);
         }
         for (const repoName of repoDependencies) {
-            relativePath = path.join('..', repoName, pluginName);
-            if (fs.existsSync(relativePath)) {
-                return relativePath;
-            }
-            // Resolve main Repository to folder called cplace
-            if (repoName === 'main') {
-                const repoAlternative = 'cplace';
-                relativePath = path.join('..', repoAlternative, pluginName);
-                if (fs.existsSync(relativePath)) {
-                    return relativePath;
-                }
+            const pathToRepo = path.resolve(repositoryDir, '..', repoName);
+            relativePathToPlugin = path.join('..', repoName, pluginName);
+            if (AssetsCompiler.isPluginFolder(pathToRepo, pluginName)) {
+                return relativePathToPlugin;
             }
         }
         console.error(cerr`Could not locate plugin ${pluginName}`);
         throw Error(`Could not locate plugin ${pluginName}`);
     }
 
+    private static isPluginFolder(repoPath: string, pluginName: string): boolean {
+        if (fs.existsSync(path.join(repoPath, "build.gradle")) &&
+            fs.existsSync(path.join(repoPath, pluginName, "build.gradle"))) {
+            return true;
+        } else if (!fs.existsSync(path.join(repoPath, "build.gradle")) &&
+            fs.existsSync(path.join(repoPath, pluginName))) {
+            return true;
```

Comment:
Maybe we should add comments explaining the situations we recognize here:
```suggestion
        if (fs.existsSync(path.join(repoPath, "build.gradle")) &&
            fs.existsSync(path.join(repoPath, pluginName, "build.gradle"))) {
            // a plugin in cplace 5.4 and later, with Gradle build files whose existence we just checked
            return true;
        } else if (!fs.existsSync(path.join(repoPath, "build.gradle")) &&
            fs.existsSync(path.join(repoPath, pluginName))) {
            // a potential plugin in cplace 5.3 and earlier, with Ant build files and IML project files; we don't check those however
            return true;
```